### PR TITLE
Add reporting to influxBD when node fails on retrieving block

### PR DESCRIPTION
**Pull request by itegulov**
## Overview
Title says it all

### JIRA ticket:
https://rchain.atlassian.net/browse/RCHAIN-3916

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: casper/src/main/scala/coop/rchain/casper/engine/Running.scala [hunk 1]
```diff
@@ -77,21 +77,21 @@ object Running {
...
-        )
-        for {
-          _  <- requestForBlock(nextPeer, hash)
-          ts <- Time[F].currentMillis
-        } yield ((hash -> Option(modifiedRequested(ts))))
-      } else {
-        val warnMessage = s"Could not retrieve requested block ${PrettyPrinter.buildString(hash)}. " +
-          "Removing the request from the requested blocks list. Casper will have to re-request the block."
-        Log[F].warn(warnMessage).as((hash -> none[Requested]))
+      requested.waitingList match {
```
Comment (kayvank) [inline casper/src/main/scala/coop/rchain/casper/engine/Running.scala:84]: this is much easier to follow & read than the previous version :)

**Review by kayvank — APPROVED** [review]

**Comment by nzpr** [comment]
bors try

File: casper/src/main/scala/coop/rchain/casper/engine/Running.scala [hunk 2]
```diff
@@ -77,21 +81,22 @@ object Running {
...
+        case nextPeer :: waitingListTail =>
+          def modifiedRequested(ts: Long) = requested.copy(
```
Comment (marcin-rzeznicki) [inline casper/src/main/scala/coop/rchain/casper/engine/Running.scala:86]: I am a bit against defining a nested function so deep into the outer function's code. Not a deal-breaker by any means, but you might reconsider moving it closer to the top

Comment (marcin-rzeznicki) [reply casper/src/main/scala/coop/rchain/casper/engine/Running.scala:86]: I am also not sure if it should be a function - it's quite obvious what the code does, there is no extra logic that would warrant an encapsulation IMO. Maybe reconsider inlining it?

**Review by marcin-rzeznicki — APPROVED** [review]
Approved - short and sweet. But I left a comment - not critical, but you might want to address it

**Pull request merged**